src/committee/orchestrator.rs

```diff
@@ -75,79 +83,120 @@ impl Orchestrator {
 
-        // pick a threshold of members at random
-        // TODO: AT RANDOM!
-        let threshold_of_members = self
-            .committee_cfg
-            .members
-            .iter()
-            .take(self.committee_cfg.threshold)
-            .collect_vec();
+        all_members.shuffle(&mut rand::thread_rng());
+        let mut stream = tokio_stream::iter(&all_members);
+
+        // Sending requests to the selected members concurrently
+        loop {
+            let (member_id, member) = match stream.next().await {
+                Some(item) => item,
+                None => break,
+            };
+            // stop now when reached enough responses
+            {
+                if results.lock().await.len() >= self.committee_cfg.threshold {}
+            }
 
-        // TODO: do this concurrently with async
-        // TODO: take a random sample instead of the first `threshold` members
-        // TODO: what if we get a timeout or can't meet that threshold? loop? send to more members?
-        for (member_id, member) in &threshold_of_members {
             // send json RPC request
             let rpc_ctx = RpcCtx::new(Some("2.0"), None, Some(member.address.clone()), None, None);
-            let resp = json_rpc_request(
+            let round1_send_json_rpc_resp = match json_rpc_request(
                 &rpc_ctx,
                 "round_1_signing",
                 &[serde_json::value::to_raw_value(&bob_request).unwrap()],
             )
             .await
-            .context("rpc request to committee didn't work");
-            debug!("{:?}", resp);
-            let resp = resp?;
-
-            let response: bitcoincore_rpc::jsonrpc::Response = serde_json::from_str(&resp)?;
-            let resp: Round1Response = response.result()?;
+            {
+                Ok(x) => x,
+                Err(err) => {
+                    info!("Error: member {member_id:?} in round 1 sending json rpc request: {err}");
+                    continue;
+                }
+            };
+
+            // parse the response json rpc type
+            let round1_parse_json_rpc_response: bitcoincore_rpc::jsonrpc::Response =
+                match serde_json::from_str(&round1_send_json_rpc_resp) {
+                    Ok(x) => x,
+                    Err(err) => {
+                        info!(
+                        "Error: member {member_id:?} in round 1 parsing json rpc response: {err}"
+                    );
+                        continue;
+                    }
+                };
+
+            // parse the actual response
+            let round1_parse_actual_response: Round1Response = match round1_parse_json_rpc_response
+                .result()
+            {
+                Ok(x) => x,
+                Err(err) => {
+                    info!("Error: member {member_id:?} in round 1 parsing actual response: {err}");
+                    continue;
+                }
+            };
```

Comment:
btw we can use `and_then` instead of all these match statements to chain `Result` types, the problem is that the error type has to be the same all the time. Something like:

```rust
let res = resp.and_then(|r| serde_json::from_str::<bitcoincore_rpc::jsonrpc::Response>(r)).and_then(|r| r.result::<Round1Response>());
match res {
   Ok(resp) => results.lock().await.push(...),
   Err(e) => /* log err */
};
```